### ISSUE TITLE
Improve dev docker experience

### ISSUE THREAD

**Issue by jonespm** [tl-its-umich-edu/my-learning-analytics#121]
As noted on another ticket by @pushyamig it would be convenient if we could install dev utilities easier. I don't think it's necessary to include them in every image though. But having an easy command/alias that can be run to install them would be nice. 

The ones that come to mind are

vim, mysql-client, silversearcher-ag

There may be some others . . .

**Comment by jonespm** [comment]
This has changed a bit, I think we'd like a better experience for development in docker. This includes
- Mapping the local file system
- Reloading changes dynamically
- Allowing for debugging in development (maybe with Werkzeug or runserver_plus or ptvsd for those using VSCode)

**Comment by jonespm** [comment]
This is going to require some research. Assigning to @DanielDugan to look at this sometime.

**Comment by DanielDugan** [comment]
Will review and provide a solution. I agree that automatic rebuilds will be valuable in dev environments.

**Comment by jonespm** [comment]
Ok, I liked your suggestion but I got pushback from ops when I implemented it.
https://github.com/tl-its-umich-edu/my-learning-analytics/pull/77

Ops said we can only go from a "Base Image" that's already published in Docker Hub, we cannot curate our own partial images (even if it's going to save us some time). Docker caches steps and doesn't run the steps that haven't changed. Openshift has a different (smaller) subset of images than Docker Hub has.

That said we could "potentially" have a different Dockerfile for dev verses production, but I was trying to avoid this. In Student Explorer the initial pass of converting to Docker I had two separate ones but I combined them because they were 99% the same there. 

docker-compose is only used for the local build so we can make any changes there.
